test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -1112,6 +1115,7 @@ public void execute(BuildContext context) {
                             .build();
 
                     List<String> testClassBeans = new ArrayList<>();
+                    testClassBeans.add(ClassloaderApplicationClassPredicate.class.getName());
```

Comment:
Can you explain the need for this? It's evident why it's necessary

Reply:
We need the predicate to be able to distinguish the application classes from fwk/lib classes. This is an implementation bean that leverages the `QuarkusClassLoader`. We could use `AdditionalBeanBuildItem` as well but since all ``TestClassBeanBuildItem``s are transformed into ``AdditionalBeanBuildItem``s it's easier... I can add a comment though ;-)

Reply:
OK, it is kinda sketchy so it at least needs to be properly commented :)

Reply:
+1 it's "sketchy" and the impl is "hacky" ;-)